utils/database_utils/mongo_utils/queries/dao/abc.py

```diff
@@ -0,0 +1,176 @@
+# NEON AI (TM) SOFTWARE, Software Development Kit & Application Framework
+# All trademark and other rights reserved by their respective owners
+# Copyright 2008-2022 Neongecko.com Inc.
+# Contributors: Daniel McKnight, Guy Daniels, Elon Gasper, Richard Leeds,
+# Regina Bloomstine, Casimiro Ferreira, Andrii Pernatii, Kirill Hrymailo
+# BSD-3 License
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from abc import ABC, abstractmethod
+
+from neon_sftp import NeonSFTPConnector
+
+from utils.database_utils import DatabaseController
+from utils.database_utils.mongo_utils import (
+    MongoQuery,
+    MongoCommands,
+    MongoFilter,
+    MongoLogicalOperators,
+)
+
+
+class MongoDocumentDAO(ABC):
+    def __init__(
+        self,
+        db_controller: DatabaseController,
+        sftp_connector: NeonSFTPConnector = None,
+    ):
+        self.db_controller = db_controller
+        self.sftp_connector = sftp_connector
+
+    @property
+    @abstractmethod
+    def document(self):
+        pass
+
+    def list_contains(
+        self,
+        key: str = "_id",
+        source_set: list = None,
+        aggregate_result: bool = True,
+        *args,
+        **kwargs
+    ) -> dict:
```

Comment:
Please add a docstring like the below method

Reply:
> Please add a docstring like the below method

please see https://github.com/NeonGeckoCom/pyklatchat/pull/79/commits/a94d596b9c878d8eef5c82d24d8546fcb4223211